plugin/documents.py

```diff
@@ -246,11 +258,11 @@ def on_reload_async(self) -> None:
     # --- Callbacks from Sublime Text ----------------------------------------------------------------------------------
 
     def on_load_async(self) -> None:
-        if self._is_regular_view():
+        if not self._text_change_listener and is_regular_view(self.view):
             self._register_async()
 
     def on_activated_async(self) -> None:
-        if self._is_regular_view() and not self.view.is_loading():
+        if not self._text_change_listener and not self.view.is_loading() and is_regular_view(self.view):
```

Comment:
Did you mean to check that the real object is still alive here or just that weakref was created? Because you are doing the latter but if that's the intention, we could make it clearer and less error-prone with a boolean.